src/coreclr/jit/emitxarch.cpp

```diff
@@ -1251,6 +1252,15 @@ emitter::code_t emitter::AddEvexPrefix(instruction ins, code_t code, emitAttr at
     return code;
 }
 
+emitter::code_t emitter::AddEvexbBit(const instrDesc* id, code_t code)
+{
+    if (hasEvexPrefix(code) && id->idIsEvexbContext())
+    {
+        code |= EVEX_B_BIT;
+    }
```

Comment:
This method should only be called by instructions which know the bit needs to be set and so should:
```
assert(IsEvexEncodableInstruction(ins));
assert((code & EVEX_B_BIT) == 0);
```

For conditionally adding the bit, we should have `AddEvexbBitIfNeeded`, to better match how other functions work.